erpnext/bulk_transaction/doctype/bulk_transaction_log/bulk_transaction_log.py

```diff
@@ -0,0 +1,58 @@
+# Copyright (c) 2021, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+import frappe
+from frappe.model.document import Document
+
+from erpnext.utilities.bulk_transaction import task, update_logger
+
+
+class BulkTransactionLog(Document):
+    pass
+
+
+@frappe.whitelist()
+def retry_failing_transaction(log_date):
+    btp = frappe.qb.DocType("Bulk Transaction Log Detail")
+    data = (
+        frappe.qb.from_(btp)
```

Comment:
This file appears to be using spaces. We use tabs everywhere. Mixing them can cause failures.

Reply:
yes I will change it

Reply:
Done